general/g.tempfile/main.c

```diff
@@ -63,8 +71,12 @@ int main(int argc, char *argv[])
     tempfile = G_tempfile_pid(p);
 
     /* create tempfile so next run of this program will create a unique name */
-    if (!dry_run->answer)
-        close(creat(tempfile, 0666));
+    if (!dry_run->answer) {
+        if (directory->answer)
+            mkdir(tempfile, 0777);
+        else
+            close(creat(tempfile, 0666));
```

Comment:
If you want this to be robust, mkdir usage in `lib/gis/` may provide good starting point.

Reply:
...also you need to, to resolve `error: too many arguments to function 'mkdir'` on Windows.

Reply:
Using `G_mkdir`now, which takes only the name as an argument...